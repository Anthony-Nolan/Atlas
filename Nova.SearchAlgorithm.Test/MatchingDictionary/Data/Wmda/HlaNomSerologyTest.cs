using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using Nova.SearchAlgorithm.MatchingDictionary.Models.Wmda;
using NUnit.Framework;

namespace Nova.SearchAlgorithm.Test.MatchingDictionary.Data.Wmda
{
    [TestFixtureSource(typeof(SerologyTestFixtureArgs), "Args")]
    [UseReporter(typeof(DiffReporter))]
    public class HlaNomSerologyTest : WmdaDataTestBase<HlaNom>
    {
        public HlaNomSerologyTest(Func<IWmdaHlaTyping, bool> filter, IEnumerable<string> matchLoci)
            : base(filter, matchLoci)
        {
        }

        [Test]
        public void HlaNomRegexCapturesSerologyAsExpected()
        {
            var oneDigit = new HlaNom("DQ", "1");
            var twoDigit = new HlaNom("A", "29");
            var threeDigit = new HlaNom("B", "703");
            var fourDigit = new HlaNom("DR", "1404");
            var deletedWithIdentical = new HlaNom("Cw", "11", true, "1");

            Assert.AreEqual(oneDigit, GetSingleWmdaHlaTyping("DQ", "1"));
            Assert.AreEqual(twoDigit, GetSingleWmdaHlaTyping("A", "29"));
            Assert.AreEqual(threeDigit, GetSingleWmdaHlaTyping("B", "703"));
            Assert.AreEqual(fourDigit, GetSingleWmdaHlaTyping("DR", "1404"));
            Assert.AreEqual(deletedWithIdentical, GetSingleWmdaHlaTyping("Cw", "11"));
        }

        [Test]
        public void HlaNomContainsAllExpectedSerology()
        {
            var str = string.Join("\r\n", AllHlaTypings
                .OrderBy(s => s.WmdaLocus)
                .ThenBy(s => int.Parse(s.Name))
                .Select(s => $"{s.WmdaLocus}\t{s.Name}")
                .ToList());
            Approvals.Verify(str);
        }
    }
}
