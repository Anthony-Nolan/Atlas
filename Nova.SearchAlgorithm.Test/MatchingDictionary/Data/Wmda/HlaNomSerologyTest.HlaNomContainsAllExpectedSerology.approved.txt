A	1
A	2
A	3
A	9
A	10
A	11
A	19
A	23
A	24
A	25
A	26
A	28
A	29
A	30
A	31
A	32
A	33
A	34
A	36
A	43
A	66
A	68
A	69
A	74
A	80
A	203
A	210
A	2403
B	5
B	7
B	8
B	12
B	13
B	14
B	15
B	16
B	17
B	18
B	21
B	22
B	27
B	35
B	37
B	38
B	39
B	40
B	41
B	42
B	44
B	45
B	46
B	47
B	48
B	49
B	50
B	51
B	52
B	53
B	54
B	55
B	56
B	57
B	58
B	59
B	60
B	61
B	62
B	63
B	64
B	65
B	67
B	70
B	71
B	72
B	73
B	75
B	76
B	77
B	78
B	81
B	82
B	703
B	2708
B	3901
B	3902
B	4005
B	5102
B	5103
Cw	1
Cw	2
Cw	3
Cw	4
Cw	5
Cw	6
Cw	7
Cw	8
Cw	9
Cw	10
Cw	11
DQ	1
DQ	2
DQ	3
DQ	4
DQ	5
DQ	6
DQ	7
DQ	8
DQ	9
DR	1
DR	2
DR	3
DR	4
DR	5
DR	6
DR	7
DR	8
DR	9
DR	10
DR	11
DR	12
DR	13
DR	14
DR	15
DR	16
DR	17
DR	18
DR	103
DR	1403
DR	1404